### PR TITLE
deploy: show changed files in PR

### PR DESCRIPTION
**Pull request by heerener**

### CODE REVIEW

File: .github/workflows/validate-commit-message.py [hunk 1]
```diff
@@ -2,82 +2,153 @@
...
+
+    commit_message_issues = []
+
+    commit = next(repo.iter_commits())
+    print(f"Checking commit: {commit.message} (parents: {commit.parents})")
+    quoted_commit_message = textwrap.indent(commit.message, prefix="  > ")
+    prefixes = collect_prefixes()
+
+    unmentioned_packages = get_unmentioned_packages(prefixes, changed_files)
+    if unmentioned_packages:
```
Comment (matz-e) [inline .github/workflows/validate-commit-message.py]: Devil's advocate: if we mass-update the copyright, this is going to blow up :D

Comment (heerener) [reply .github/workflows/validate-commit-message.py]: Very good point - maybe we should temporarily turn this off for that kind of changes 🥷

Comment (matz-e) [reply .github/workflows/validate-commit-message.py]: Or define a custom threshold like 3 or 5 packages changed total?

File: .github/workflows/validate-commit-message.py [hunk 2]
```diff
@@ -2,82 +2,154 @@
...
+        commit_message_issues.append(msg)
+
+    if (
+        any("documentation" in changed_file for changed_file in changed_files)
+        and "docs" not in prefixes
+    ):
+        msg = textwrap.dedent(
+            """\
+            * Docs were changed but not mentioned in the commit message.
+              Please use the docs: prefix to explain this change.
```
Comment (matz-e) [inline .github/workflows/validate-commit-message.py]:
```suggestion
              Please use the `docs:` prefix to explain this change.
```

File: .github/workflows/validate-commit-message.py [hunk 3]
```diff
@@ -2,82 +2,154 @@
...
+        )
+        commit_message_issues.append(msg)
+
+    deploy_changed = (any("yml" in changed_file for changed_file in changed_files) or
+                      any("yaml" in changed_file for changed_file in changed_files))
+    if deploy_changed and "deploy" not in prefixes:
+        msg = textwrap.dedent(
+            """\
+            * Deploy files were changed but not mentioned in the commit message.
+              Please use the deploy: prefix to explain this change.
```
Comment (matz-e) [inline .github/workflows/validate-commit-message.py]:
```suggestion
            * Deployment files were changed but not mentioned in the commit message.
              Please use the `deploy:` prefix to explain this change.
```

File: .github/workflows/validate-commit-message.py [hunk 4]
```diff
@@ -2,82 +2,154 @@
...
+    prefixes = collect_prefixes(commit.message)
+
+    unmentioned_packages = get_unmentioned_packages(prefixes, changed_files)
+    if unmentioned_packages:
+        msg = textwrap.dedent(
+            f"""\
+            * The following packages were changed but not mentioned:
+              {", ".join(unmentioned_packages)}
+              You can simply use the above list followed by a colon, then explain what you changed.
+              Alternatively, you can use a line per package to describe the change per package.
```
Comment (matz-e) [inline .github/workflows/validate-commit-message.py]:
```suggestion
              Alternatively, you can use one line per package to describe the changes per package.
```

File: .github/workflows/validate-commit-message.py [hunk 5]
```diff
@@ -2,82 +2,175 @@
...
+    print(f"Checking commit: {commit.message} (parents: {commit.parents})")
+    quoted_commit_message = textwrap.indent(commit.message, prefix="  > ")
+
+    prefixes = collect_prefixes(title)
+    docs_changed = any(
+        "documentation" in changed_file for changed_file in changed_files
+    )
+    deploy_changed = any(
+        "yml" in changed_file for changed_file in changed_files
+    ) or any("yaml" in changed_file for changed_file in changed_files)
```
Comment (matz-e) [inline .github/workflows/validate-commit-message.py]:
```suggestion
    deploy_changed = any(
        changed_file.ends_with(".yml") or changed_file.ends_with(".yaml")
        for changed_file in changed_files
    )
```

File: .github/workflows/validate-commit-message.py [hunk 6]
```diff
@@ -2,82 +2,175 @@
...
+
+    minimal_prefix_present = False
+    if one_package_mentioned(prefixes, changed_files):
+        minimal_prefix_present = True
+    elif docs_changed and "docs" in prefixes:
+        minimal_prefix_present = True
+    elif deploy_changed:
```
Comment (matz-e) [inline .github/workflows/validate-commit-message.py]: Shouldn't this also end with `"deploy" in prefixes`?

**Review by matz-e — APPROVED** [review]

**Pull request merged**